pkg/release/release_version.go

```diff
@@ -0,0 +1,276 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package release
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/release/pkg/git"
+	"k8s.io/release/pkg/release/regex"
+)
+
+const (
+	ReleaseTypeOfficial string = "official"
+	ReleaseTypeRC       string = "rc"
+	ReleaseTypeBeta     string = "beta"
+	ReleaseTypeAlpha    string = "alpha"
+)
+
+// Versions specifies the collection of found release versions
+type Versions struct {
+	prime    string
+	official string
+	rc       string
+	beta     string
+	alpha    string
+}
+
+// Prime can be used to get the most prominent release version
+func (r *Versions) Prime() string {
+	return r.prime
+}
+
+// Official can be used to get the ReleaseTypeOfficial from the versions
+func (r *Versions) Official() string {
+	return r.official
+}
+
+// Rc can be used to get the ReleaseTypeRC from the versions
+func (r *Versions) RC() string {
+	return r.rc
+}
+
+// Beta can be used to get the ReleaseTypeBeta from the versions
+func (r *Versions) Beta() string {
+	return r.beta
+}
+
+// Alpha can be used to get the ReleaseTypeAlpha from the versions
+func (r *Versions) Alpha() string {
+	return r.alpha
+}
+
+// String returns a string representation for the release versions
+func (r *Versions) String() string {
```

Comment:
Newish to Go but for these two methods is something like this preferred?

```
switch n := ""; n {
case r.official != n: 
              fmt.Fprintf(sb, "%s: %s, ", ReleaseTypeOfficial, r.official)
              fallthrough
/
res = append(res, r.official)
case...
}
```
Just curious

Reply:
Yeah that would be possible too for sure. From my point of view the current implementation seems a bit easier to read. We could also check for `len(r.official)` but all in all it does not matter much. :) Thanks for the input, though.